### PR TITLE
Make AMQP provisioning call end if connection is lost partway through

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Same strategy we took in MQTT. Link the user's cancellation token to a cancellation token that is cancelled when the connection is lost. Then be careful to throw ProvisioningDeviceException if only the connection loss token was cancelled

### CODE REVIEW

**Review by drwill-ms — APPROVED** [review]

File: provisioning/device/src/Transports/Amqp/AmqpClientConnection.cs [hunk 1]
```diff
@@ -151,7 +151,10 @@ public AmqpClientSession CreateSession()
 
         private void OnConnectionClosed(object o, EventArgs args)
         {
-            IsConnectionClosed = true;
```
Comment (timtay-microsoft) [inline provisioning/device/src/Transports/Amqp/AmqpClientConnection.cs:154]: This flag wasn't used anywhere

File: provisioning/device/src/Transports/Amqp/ProvisioningTransportHandlerAmqp.cs [hunk 2]
```diff
@@ -24,10 +24,10 @@ public class ProvisioningTransportHandlerAmqp : ProvisioningTransportHandler
         // DPS will generally send a retry-after header that overrides this default value though.
         private static readonly TimeSpan s_defaultOperationPollingInterval = TimeSpan.FromSeconds(2);
 
-        private static readonly TimeSpan s_timeoutConstant = TimeSpan.FromMinutes(1);
```
Comment (timtay-microsoft) [inline provisioning/device/src/Transports/Amqp/ProvisioningTransportHandlerAmqp.cs:27]: We no longer need this timeout. Users can just pass in a cancellation token

**Pull request merged**